### PR TITLE
Adding new repository for indexing: rclada

### PR DESCRIPTION
**Pull request by mosteo**
Adding Ada client library main repository for initial documentation indexing.

### CODE REVIEW

**Comment by mosteo** [comment]
I'm reading in the review guidelines:

"The source repository should contain one or more ROS packages (meaning they have a package.xml in the source repository)"

The repository I intend to submit has the packages as submodules. Is this acceptable? Thanks.

**Comment by ivanpauno** [comment]
> The repository I intend to submit has the packages as submodules. Is this acceptable? Thanks.

AFAIK we don't have any policy for that.
It sounds fine to me, submodules are automatically cloned when cloning the repo, but IDK what others think.

@nuclearsandwich what do you think?

File: foxy/distribution.yaml [hunk 1]
```diff
@@ -2673,6 +2673,12 @@ repositories:
       url: https://github.com/ros2/rcl_logging.git
       version: foxy
     status: maintained
+  rclada:
+    doc:
```
Comment (ivanpauno) [inline foxy/distribution.yaml:2677]: I would add a `src` entry as well

File: foxy/distribution.yaml [hunk 2]
```diff
@@ -2673,6 +2673,12 @@ repositories:
       url: https://github.com/ros2/rcl_logging.git
       version: foxy
     status: maintained
+  rclada:
```
Comment (ivanpauno) [inline foxy/distribution.yaml:2676]: shouldn't this match the repo name?

File: foxy/distribution.yaml [hunk 3]
```diff
@@ -2673,6 +2673,12 @@ repositories:
...
+    doc:
+      type: git
+      url: https://github.com/ada-ros/ada4ros2.git
+      version: foxy
```
Comment (clalancette) [inline foxy/distribution.yaml:2680]: There's trailing whitespace at the end of this line, which is why CI is failing.

**Comment by clalancette** [comment]
> @nuclearsandwich what do you think?
> 
> edit: if this repo will have an entry, I think than entries for the submodules shouldn't be added, but IDK.

This setup isn't really how the buildfarm works.

What you seem to have here is a "meta" repository that contains all of the other repositories, plus some scripts and other things.  But the buildfarm here really only deals in repositories with one or more packages in it.  So I don't think this is a good way to go about this.

Instead, I'll suggest adding one entry for each of the submodule entries, i.e. https://github.com/ada-ros/rosidl_generator_ada, https://github.com/ada-ros/rclada_common , etc.  That matches much more closely how everything else in this file is setup.

**Comment by ivanpauno** [comment]
> Instead, I'll suggest adding one entry for each of the submodule entries, i.e. https://github.com/ada-ros/rosidl_generator_ada, https://github.com/ada-ros/rclada_common , etc. That matches much more closely how everything else in this file is setup.

:+1: that matches more closely what we do in other places.

> But the buildfarm here really only deals in repositories with one or more packages in it. So I don't think this is a good way to go about this.

I don't think the buildfarm will realize, submodules are transparent (i.e. the buildfarm will still think these are many packages in one repo). But I might be wrong and there's also no precedent of doing this.

**Comment by nuclearsandwich** [comment]
> > 
> 
> AFAIK we don't have any policy for that.
> It sounds fine to me, submodules are automatically cloned when cloning the repo, but IDK what others think.

I think it makes the best sense to add the package repositories directly. Submodule support if it exists everywhere is going to be somewhat accidental and the layer of indirection would be unexpected for most of the ROS community.

**Comment by mosteo** [comment]
Thanks for the review, I will do as suggested.

**Review by ivanpauno — APPROVED** [review]

**Comment by ivanpauno** [comment]
Thanks @mosteo !

**Pull request merged**